test/extended/setup_compat.sh

```diff
@@ -0,0 +1,354 @@
+#!/bin/bash
+#
+# Starts an OpenShift master for compatibility testing
+
+
+# Launches a server for OpenShift compatibility testing
+# PREVIOUS_CONTROLLER=1 launches with previous version controllers
+# PREVIOUS_API=1        launches with previous version API server
+
+# Sets up a current master server with previous controllers
+function os::test::extended::setup_compatibility {
+  # os::log::stacktrace::install
+
+  # build binaries
+  if [[ -z $(os::build::find-binary ginkgo) ]]; then
+    hack/build-go.sh vendor/github.com/onsi/ginkgo/ginkgo
+  fi
+  if [[ -z $(os::build::find-binary extended.test) ]]; then
+    hack/build-go.sh test/extended/extended.test
+  fi
+  if [[ -z $(os::build::find-binary openshift) ]]; then
+    hack/build-go.sh
+  fi
+  if [[ -z "${PREVIOUS_VERSION:-}" ]]; then
+    if [[ -z $(os::build::find-binary previousversion) ]]; then
+      hack/build-go.sh tools/previousversion
+    fi
+    openshift_binary=$(os::build::find-binary openshift)
+    previous_version=$(os::build::find-binary previousversion)
+    export PREVIOUS_VERSION="$(${previous_version} ${openshift_binary})"
+  fi
+  if [[ -z "${API_SERVER_VERSION:-}" ]] && [[ -n "${PREVIOUS_API:-}" ]]; then
+    API_SERVER_VERSION="${PREVIOUS_VERSION}"
+  fi
+  if [[ -z "${CONTROLLER_VERSION:-}" ]] && [[ -n "${PREVIOUS_CONTROLLER:-}" ]]; then
+    CONTROLLER_VERSION="${PREVIOUS_VERSION}"
+  fi
+
+  os::util::environment::setup_time_vars
+
+  # ensure proper relative directories are set
+  export GINKGO="$(os::build::find-binary ginkgo)"
+  export EXTENDEDTEST="$(os::build::find-binary extended.test)"
+  export TMPDIR=${BASETMPDIR:-/tmp}
+  export EXTENDED_TEST_PATH="$(pwd)/test/extended"
+  export KUBE_REPO_ROOT="$(pwd)/vendor/k8s.io/kubernetes"
+
+  # output tests instead of running
+  if [[ -n "${SHOW_ALL:-}" ]]; then
+    TEST_OUTPUT_QUIET=true ${EXTENDEDTEST} --ginkgo.dryRun --ginkgo.noColor | grep ok | grep -v skip | cut -c 20- | sort
+    exit 0
+  fi
+
+  # ensure_iptables_or_die
+
+  function cleanup()
+  {
+    out=$?
+    cleanup_openshift
+    echo "[INFO] Exiting"
+    return $out
+  }
+
+  trap "exit" INT TERM
+  trap "cleanup" EXIT
+  echo "[INFO] Starting server"
+
+  os::util::environment::setup_all_server_vars "test-extended/compatibility"
+  os::util::environment::use_sudo
+  os::util::environment::setup_images_vars
+  reset_tmp_dir
+
+  # If the current system has the XFS volume dir mount point we configure
+  # in the test images, assume to use it which will allow the local storage
+  # quota tests to pass.
+  if [ -d "/mnt/openshift-xfs-vol-dir" ]; then
```

Comment:
also do we want to be doing this check or using `lvs`, since the check we *really* want is that there is the backing storage for the mount point, and this isn't a strong enough check?